Source/ASImageNode+AnimatedImage.mm

```diff
@@ -53,6 +53,14 @@ - (void)_locked_setAnimatedImage:(id <ASAnimatedImageProtocol>)animatedImage
   if (ASObjectIsEqual(_animatedImage, animatedImage)) {
     return;
   }
+    
+  if (animatedImage == nil) {
+    // Animated image can take while to dealloc, do it off the main queue
+    __block id <ASAnimatedImageProtocol>deallocImage = _animatedImage;
+    ASPerformBlockOnBackgroundThread(^{
```

Comment:
For consistency, let's use `ASPerformBackgroundDeallocation()`. Also, let's use `previousAnimatedImage` declared a few lines below.

Reply:
Lastly, I think it's easier to reason if we do this at the end of this method.